frontend/src/metabase/common/components/Table/Table.tsx

```diff
@@ -29,102 +25,56 @@ export function Table<Row extends BaseRow>({
   columns,
   rows,
   rowRenderer,
-  ...tableProps
+  tableProps,
+  formatValueForSorting = (row: Row, columnName: string) => row[columnName],
+  locale,
+  ...rest
 }: TableProps<Row>) {
   const [sortColumn, setSortColumn] = React.useState<string | null>(null);
   const [sortDirection, setSortDirection] = React.useState<"asc" | "desc">(
     "asc",
   );
 
-  const sortedRows = React.useMemo(() => {
+  const compareStrings = useCallback(
+    (a: string, b: string) =>
+      a.localeCompare(b, locale, { sensitivity: "base" }),
+    [locale],
+  );
+
+  const sortedRows = useMemo(() => {
     if (sortColumn) {
```

Comment:
Maybe it could become `const compareStrings = useCompareStrings()` and then that hook could look up the locale and return the callback – to keep concerns separated

Reply:
A hook that looks up the locale would go back to the table depending on the redux store and all that fun stuff. For the time being, I moved it outside the component and just had it accept the `locale` as an optional prop.

For what it's worth, I think the real answer here is to expose the sorting in a way that it actually exists outside of the (now) `ClientSortableTable` component. And there, if you want to compare by locale, it's very easy to do so. But one of the goals of this PR is to make rebasing as seamless as possible for both @rafpaf and myself, so I'm inclined to leave the API as it is for now and adjust in the future if we decide to